InnerEye/ML/deep_learning_config.py

```diff
@@ -254,6 +263,50 @@ def validate(self) -> None:
                              f"found number_of_cross_validation_splits = {self.number_of_cross_validation_splits} "
                              f"and cross_validation_split_index={self.cross_validation_split_index}")
 
+    """ Defaults for when to run inference in the absence of any command line switches. """
+    INFERENCE_DEFAULTS: Dict[ModelProcessing, Dict[ModelExecutionMode, bool]] = {
+        ModelProcessing.DEFAULT: {
+            ModelExecutionMode.TRAIN: False,
+            ModelExecutionMode.TEST: True,
+            ModelExecutionMode.VAL: True,
+        },
+        ModelProcessing.ENSEMBLE_CREATION: {
+            ModelExecutionMode.TRAIN: False,
+            ModelExecutionMode.TEST: True,
+            ModelExecutionMode.VAL: False,
+        }
+    }
+
+    """ Mapping from ModelProcesing and ModelExecutionMode to command line switch. """
+    INFERENCE_OPTIONS: Dict[ModelProcessing, Dict[ModelExecutionMode, str]] = {
+        ModelProcessing.DEFAULT: {
+            ModelExecutionMode.TRAIN: 'inference_on_train_set',
+            ModelExecutionMode.TEST: 'inference_on_test_set',
+            ModelExecutionMode.VAL: 'inference_on_val_set',
+        },
+        ModelProcessing.ENSEMBLE_CREATION: {
+            ModelExecutionMode.TRAIN: 'ensemble_inference_on_train_set',
+            ModelExecutionMode.TEST: 'ensemble_inference_on_test_set',
+            ModelExecutionMode.VAL: 'ensemble_inference_on_val_set',
+        }
+    }
+
```

Comment:
Is there enough test coverage to ensure that all those string names are really valid attribute names?

Reply:
I've reworked it to avoid the getattr. This is now a function that returns a Dict of all these parameters.